go/yaml2/yaml_test.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2024 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml2
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestYamlVars(t *testing.T) {
+	type TestStruct struct {
+		StringField  string  `yaml:"stringfield"`
+		IntField     int     `yaml:"intfield"`
+		BoolField    bool    `yaml:"boolfield"`
+		Float64Field float64 `yaml:"float64field"`
+	}
+
+	inputData := TestStruct{
+		"tricky text to test text",
+		32,
+		true,
+		3.141,
+	}
+
+	//testing Marshal
+	marshaledData, err := Marshal(inputData)
+	assert.NoError(t, err)
+
+	//testing Unmarshal
+	var unmarshaledData TestStruct
+	err = Unmarshal(marshaledData, &unmarshaledData)
+	assert.NoError(t, err)
+	if !reflect.DeepEqual(inputData, unmarshaledData) {
```

Comment:
`assert.Equal` afaik ends up doing a deep equal too, so let's use that.

Reply:
Yeah [this](https://github.com/stretchr/testify/blob/v1.8.4/assert/assertions.go#L58) does use deep equal. Will change it.